File: AdminArea/AdminArea.Web/Modules/onlinestore/products/productsForm.cs
using Serenity.ComponentModel;
using System;

namespace AdminArea.onlinestore.Forms;

[FormScript("onlinestore.products")]
[BasedOnRow(typeof(productsRow), CheckNames = true)]
public class productsForm
{
    public string Name { get; set; }
    public decimal Price { get; set; }
    public string Description { get; set; }
    public int Code { get; set; }
    public int CategoryId { get; set; }
    public string Photo { get; set; }
    public string SupplierName { get; set; }
    public DateTime EntryDate { get; set; }
    public string ReviewUrl { get; set; }
    public int Quantity { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/onlinestore/productsForm.ts
import { StringEditor, DecimalEditor, IntegerEditor, LookupEditor, ImageUploadEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface productsForm {
    Name: StringEditor;
    Price: DecimalEditor;
    Description: StringEditor;
    Code: IntegerEditor;
    CategoryId: LookupEditor;
    Photo: ImageUploadEditor;
    SupplierName: StringEditor;
    EntryDate: DateEditor;
    ReviewUrl: StringEditor;
    Quantity: IntegerEditor;
}

export class productsForm extends PrefixedContext {
    static formKey = 'onlinestore.products';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!productsForm.init)  {
            productsForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = IntegerEditor;
            var w3 = LookupEditor;
            var w4 = ImageUploadEditor;
            var w5 = DateEditor;

            initFormType(productsForm, [
                'Name', w0,
                'Price', w1,
                'Description', w0,
                'Code', w2,
                'CategoryId', w3,
                'Photo', w4,
                'SupplierName', w0,
                'EntryDate', w5,
                'ReviewUrl', w0,
                'Quantity', w2
            ]);
        }
    }
}